### PR TITLE
[FLINK-23352][streaming-java] Add configuration for batch size and socket timeout of collect sink

### PR DESCRIPTION
**Pull request by tsreaper**
## What is the purpose of the change

Collect sink has a batch size which is currently not configurable. By the user request in FLINK-23352 we're making it configurable in this PR to support user with super large row size.

Note that these configuration are only available for table APIs. As we cannot read user config in data stream API, data stream users will have to write their own Java code to create collect sink explicitly if they want to configure these.

## Brief change log

 - Add configuration for batch size and socket timeout of collect sink

## Verifying this change

This change added tests and can be verified by running the added tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/runtime/batch/sql/TableSinkITCase.scala [hunk 1]
```diff
@@ -76,4 +81,22 @@ class TableSinkITCase extends BatchTestBase {
...
+    tEnv.getConfig.getConfiguration.set(
+      CollectSinkOperatorFactory.MAX_BATCH_SIZE, MemorySize.parse("1b"))
+    try {
+      checkResult("SELECT 1", Seq(row(1)))
+      Assert.fail("Expecting exception thrown from collect sink")
+    } catch {
+      case e: Exception =>
+        MatcherAssert.assertThat(
+          e,
+          FlinkMatchers.containsMessage("Record size is too large for CollectSinkFunction"))
```
Comment (wuchong) [inline flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/runtime/batch/sql/TableSinkITCase.scala:95]: Could you also tell users how to change the configuretion and the config name in the exception message?

**Review by wuchong — APPROVED** [review]
LGTM.

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/runtime/batch/sql/TableSinkITCase.scala (hunk 1 above)
Comment (YasuoStudyJava) [reply flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/runtime/batch/sql/TableSinkITCase.scala]: > Could you also tell users how to change the configuretion and the config name in the exception message?

Thank you for above comments，but I still hava a question. Is there a maximum limit on this value? If users can configure it as much as possible, there should be a maximum limit, such as not exceeding the size of taskmanager's taskmanager. memory. process. size？ 
Thank you for your reply！

**Comment by YasuoStudyJava** [comment]
Thank you for above comments，but I still hava a question. Is there a maximum limit on this value? If users can configure it as much as possible, there should be a maximum limit, such as not exceeding the size of taskmanager's taskmanager. memory. process. size？
Thank you for your reply！

**Pull request merged**